whylog_vim/gui.py

```diff
@@ -0,0 +1,209 @@
+        vim.command('setlocal nomodifiable')
+
+
+class VimEditor():
+
+    whylog_windows = [
+        WindowTypes.QUERY,
+        WindowTypes.TEACHER,
+        WindowTypes.INPUT,
+        WindowTypes.CASE,
+    ]
+
+    def get_input_content(self):
+        with self.output_window_context as output_buffer:
+            return filter_comments(output_buffer[:])
+
+    def go_to_file(self, file_name, offset=1):
+        id_ = self.get_files_window_id(file_name)
+        if id_ is not None:
+            self.go_to_window(id_)
+            self.go_to_offset(offset)
+        else:
+            self.split_window()
+            self.open_file_at_offset(file_name, offset)
+
+    def create_case_window(self, default_input):
+        self.output_window_context = WindowContext(self)
+        vim.command(':e %s' % WindowTypes.CASE)
+        vim.command(':setlocal buftype=nowrite')
+        self.set_output(default_input)
+        vim.command(':set nomodifiable')
+        self._set_file_type()
+
+    def create_input_window(self, default_input):
+        self.output_window_context = WindowContext(self)
+        vim.command(':e %s' % WindowTypes.INPUT)
+        vim.command(':setlocal buftype=nowrite')
+        self.set_output(default_input)
+        vim.command(':set modifiable')
+        self._set_file_type()
+
+    def create_teacher_window(self):
+        self.output_window_context = WindowContext(self)
+        vim.command(':e %s' % WindowTypes.TEACHER)
+        vim.command(':setlocal buftype=nowrite')
+        vim.command(':set nomodifiable')
+        self._set_file_type()
+
+    def create_query_window(self):
+        self.output_window_context = WindowContext(self)
+        vim.command(':rightbelow split %s' % WindowTypes.QUERY)
+        vim.command(':setlocal buftype=nowrite')
+        self.resize(10)
+        vim.command(':setlocal nomodifiable')
+        self._set_file_type()
+
+    def close_window(self, filename):
+        self.go_to_file(filename)
+        vim.command(':q')
+
+    def get_button_name(self):
+        return get_button_name(self.get_current_line(), self.get_col())
+
+    def set_output(self, contents):
+        with self.output_window_context as output_buffer:
+            output_buffer[:] = contents.split('\n')
+
+    def _get_output_window_id(self):
+        for id_, window in enumerate(vim.windows):
+            for name in self.whylog_windows:
+                if window.buffer.name.endswith(name):
+                    return int(id_)+1
+        return None
+
+    def get_cursor_offset(self):
+        """
+        This method return offset -
+        byte of the first char of the line
+        where is cursor, when the method is called.
+        """
+        return int(vim.eval('line2byte(line("."))'))
+
+    def get_front_input(self):
+        """
+        This method return Front Input object of the line where cursor is,
+        which is implemented in the whylog.front.utils.py file.
```

Comment:
Remove this comment.  After refactor it will be useless. It is easy to find FrontInput implementation place.